pyirf/interpolation/component_estimators.py

```diff
@@ -540,8 +563,101 @@ def __call__(self, target_point):
             effective areas. For RAD_MAX_2D of shape (n_energy_bins, n_fov_offset_bins)
 
         """
+        # First, construct estimation without handling fill-values
+        full_estimation = super().__call__(target_point)
+        # Safeguard against extreme extrapolation cases
+        full_estimation[full_estimation < 0] = 0
+
+        # Early exit if fill_values should not be handled
+        if not self.fill_val:
+            return full_estimation
+
+        # Raise error if fill-values should be handled in >=3 dims
+        if self.grid_dim >= 3:
+            raise ValueError(
+                "Fill-value handling only supported in up to two grid dimensions."
+            )
+
+        # Early exit if a nearest neighbor estimation would be overwritten
```

Comment:
Why can't we just check for nearest neighbor? Why this complex setup?

Reply:
That setup is to catch cases where the user decided to mix approaches and use e.g. NearestNeighbor for interpolation but an actual extrapolator for extrapolation or vice-versa. I'll extend the comment in code to make this more clear for the future.